src/rantanplan/structures.py

```diff
@@ -231,6 +231,32 @@ def is_ovillejo(lengths):
     return correct_lines == number_of_verses
 
 
+def count_characters(string):
+    """
+    Count how many times does a character occur in a given string before its
+    own position
+    :param string: String with the rhyme pattern
+    :return: A string with the count values
+    """
+    count_dict = {}
+    output = ""
+    for character in string:
+        count = count_dict.get(character, 0)
+        output += str(count)
+        count_dict[character] = count + 1
+    return output
```

Comment:
The one liner ICYMI:
```
from collections import defaultdict
counter = defaultdict(int)                                                                                                                   
return "".join([str(counter.update({c: counter[c] + 1}) or counter[c] - 1) for c in s])
```
I would make this function return the plain list of numbers.

Not sure the function name is descriptive enough.